packages/core/src/state-management/state-manager.ts

```diff
@@ -340,14 +346,57 @@ export class StateManager {
     return this._processAnchorResponse(state$, anchorStatus$)
   }
 
-  private async _saveAnchorRequestForState(state$: RunningState): Promise<void> {
+  private async _buildAnchorRequestCARFile(streamId: StreamID, tip: CID): Promise<CAR> {
+    const car = this.carFactory.build()
+
+    // Root block
+    const timestampISO = new Date().toISOString()
+    car.put(
+      {
+        timestamp: timestampISO,
+        streamId: streamId.bytes,
+        tip: tip,
+      },
+      { isRoot: true }
+    )
+
+    const cidToBlock = async (cid) => new CarBlock(cid, await this.dispatcher._ipfs.block.get(cid))
+
+    // Genesis block
+    const genesisCid = streamId.cid
+    car.blocks.put(await cidToBlock(genesisCid))
+
+    // Tip block
+    car.blocks.put(await cidToBlock(tip))
+
+    // Genesis Link Block
+    const genesisCommit = await this.dispatcher.retrieveCommit(genesisCid, streamId)
```

Comment:
Why not this? Worth trying to minimize the possible calls out to IPFS as they can require a http request to kubo.
```suggestion
    const genesisCommit = car.get(genesisCid)
```